### PR TITLE
Add some typespecs

### PR DESCRIPTION
**Pull request by kbaird**
Adds some typespecs

### CODE REVIEW

File: lib/jsonapi/error_view.ex [hunk 1]
```diff
@@ -34,37 +37,43 @@ defmodule JSONAPI.ErrorView do
...
 
+  @spec missing_id :: map()
   def missing_id do
     "Missing id in data parameter"
     |> build_error(400, @crud_message, "/data/id")
     |> serialize_error
   end
 
-  def to_many_relationships_payload_for_standard_endpoint do
+  @spec too_many_relationships_payload_for_standard_endpoint :: map()
```
Comment (kbaird) [inline lib/jsonapi/error_view.ex]: I also fixed a typo here

Comment (doomspork) [reply lib/jsonapi/error_view.ex]: I wonder if we should continue to define "to_many" with a deprecation notice in the event folks are using these directly?

Comment (kbaird) [reply lib/jsonapi/error_view.ex]: My mistake. https://github.com/jeregrine/jsonapi/pull/198/commits/55dd0f2bf13a61b65302fabf1570318d5c492e3e reverts.

Comment (kbaird) [reply lib/jsonapi/error_view.ex]: My confusion sprang from the fact that both labels are appropriate. It is an error that pertains to a "to-many" relationship, and is triggered by their count being too high (i.e., there are "too many" of them).

**Pull request merged**